homeassistant/components/modbus/base_platform.py

```diff
@@ -76,7 +77,7 @@ class BasePlatform(Entity):
     def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
         """Initialize the Modbus binary sensor."""
         self._hub = hub
-        self._slave = entry.get(CONF_SLAVE, 0)
+        self._slave = entry.get(CONF_SLAVE, entry.get(CONF_DEVICE_ADDRESS, 0))
```

Comment:
This is also on option, the only difference is that it will alway evaluate `entry.get(CONF_DEVICE_ADDRESS, 0)` even when it is not used.

Reply:
The `or` method will go on if there was a result.